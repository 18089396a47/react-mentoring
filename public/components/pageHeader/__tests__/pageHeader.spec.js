import React from 'react';
import renderer from 'react-test-renderer';
import PageHeader from '../pageHeader';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('PageHeader component', () => {
    it('PageHeader should be shown correctly without headerComponents', () => {
        const tree = renderer.create(
            <PageHeader>
                <span>children</span>
            </PageHeader>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('PageHeader should be shown correctly with headerComponents', () => {
        const tree = renderer.create(
            <PageHeader
                headerComponents={<span>headerComponents</span>}
                >
                <span>children</span>
            </PageHeader>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
