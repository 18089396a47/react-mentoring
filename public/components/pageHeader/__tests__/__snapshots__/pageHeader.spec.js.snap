// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component PageHeader should be shown correctly with headerComponents 1`] = `
<header
  className="page-header"
>
  <div
    className="page-header__link-wrapper"
  >
    <Link
      className="page-header__link"
      to="/"
    >
      netflixroulette
    </Link>
    <span>
      headerComponents
    </span>
  </div>
  <span>
    children
  </span>
</header>
`;

exports[`PageHeader component PageHeader should be shown correctly without headerComponents 1`] = `
<header
  className="page-header"
>
  <div
    className="page-header__link-wrapper"
  >
    <Link
      className="page-header__link"
      to="/"
    >
      netflixroulette
    </Link>
  </div>
  <span>
    children
  </span>
</header>
`;
